### PR TITLE
[#1851] part 4

### PR DESCRIPTION
**Pull request by valllllll2000**
#### The solution
Remove duplicated classes between utils and app modules
Use the utils classes inside the app module, code related to FormActivity
Moved tests to the right module (the ones related to xml)
#### Reviewer Checklist
* [ ] Added an explanation about the work done
* [ ] Connected the PR and the issue on Zenhub
* [ ] Added a test plan to the issue
* [ ] Updated the copyright header
* [ ] Formatted the code per our style guide
* [ ] Added a documentation (if relevant)

### CODE REVIEW

File: app/src/androidTest/java/org/akvo/flow/activity/form/data/SurveyInstaller.java [hunk 1]
```diff
@@ -195,7 +195,7 @@ public void deleteResponses() {
     }
 
     public SparseArray<List<Node>> getAllNodes(Question question, Context context) {
-        String src = question.getSrc();
+        String src = question.getCascadeResource();
```
Comment (muloem) [inline app/src/androidTest/java/org/akvo/flow/activity/form/data/SurveyInstaller.java:198]:
```suggestion
        String cascadeFileName = question.getCascadeResource();
```

File: utils/src/main/java/org/akvo/flow/utils/entity/Question.kt [hunk 2]
```diff
@@ -22,34 +22,97 @@ package org.akvo.flow.utils.entity
...
+        return if (isRepeatable()) {
+            val questionIdAndIteration: List<String> = questionId?.split("\\|") ?: emptyList()
+            val iteration = questionIdAndIteration[1]
+            iteration.toInt()
+        } else {
+            -1
+        }
+    }
+
+    //TODO: remove
```
Comment (muloem) [inline utils/src/main/java/org/akvo/flow/utils/entity/Question.kt:65]: Why not remove now?

File: utils/src/main/java/org/akvo/flow/utils/entity/Question.kt [hunk 3]
```diff
@@ -22,34 +22,97 @@ package org.akvo.flow.utils.entity
...
+    var cascadeResource: String? = null,
+    var levels: MutableList<Level> = mutableListOf(),
+    var validationRule: ValidationRule? = null,
 ) {
 
     fun addAltText(altText: AltText) {
         languageTranslationMap[altText.languageCode] = altText
     }
+
+    //TODO: remove
```
Comment (muloem) [inline utils/src/main/java/org/akvo/flow/utils/entity/Question.kt:54]: Why not remove now?

File: app/src/androidTest/java/org/akvo/flow/activity/form/data/SurveyInstaller.java (hunk 1 above)
Comment (valllllll2000) [reply app/src/androidTest/java/org/akvo/flow/activity/form/data/SurveyInstaller.java]: Changed

File: utils/src/main/java/org/akvo/flow/utils/entity/Question.kt (hunk 3 above)
Comment (valllllll2000) [reply utils/src/main/java/org/akvo/flow/utils/entity/Question.kt:54]: I am working on this in part 6, it's not easy to just remove

File: utils/src/main/java/org/akvo/flow/utils/entity/Question.kt (hunk 2 above)
Comment (valllllll2000) [reply utils/src/main/java/org/akvo/flow/utils/entity/Question.kt:65]: same comment as above

**Review by muloem — APPROVED** [review]

**Pull request merged**